task_processing/plugins/kubernetes/utils.py

```diff
@@ -52,7 +55,38 @@ def is_secret_env_var(value: str) -> bool:
     return SECRET_VALUE_REGEX.match(value) is not None
 
 
-def get_kubernetes_env_vars(environment: PMap[str, str]) -> List[V1EnvVar]:
+def get_secret_name_from_ref(value: str) -> str:
+    return value.split("(")[1][:-1]
+
+
+def is_shared_secret(value: str) -> bool:
```

Comment:
maybe we should add a note here for the benefit of non-Yelpers looking at this code about what shared secrets are?
e.g.,
"""
In some cases, multiple services need access to the same secrets - rather than storing these N times internally, Yelp has a mechanism in which a secret can be tagged as "shared" and thus only need to be modified in one place should that secret need to be rotated/updated/etc.

These "shared" secrets are referenced using the format `SHARED_SECRET(secret_name)` instead of the more tightly-scoped format of `SECRET(secret_name)`
"""